servicetalk-concurrent-internal/src/main/java/io/servicetalk/concurrent/internal/SubscriberUtils.java

```diff
@@ -335,4 +349,39 @@ public static void handleExceptionFromOnSubscribe(CompletableSource.Subscriber s
             LOGGER.info("Ignoring exception from onComplete of Subscriber {}.", subscriber, t);
         }
     }
+
+    /**
+     * Invokes {@link SingleSource.Subscriber#onSuccess(Object)} ignoring an occurred exception if any.
+     * @param subscriber The {@link SingleSource.Subscriber} that may throw an exception from
+     * {@link SingleSource.Subscriber#onSuccess(Object)}.
+     * @param value The value to pass to {@link SingleSource.Subscriber#onSuccess(Object)}.
+     * @param <T> The type of {@link SingleSource.Subscriber}.
+     */
+    public static <T> void safeOnComplete(SingleSource.Subscriber<T> subscriber, @Nullable T value) {
```

Comment:
The name seems misleading, perhaps you wanted to name this `deliverTerminalFromSource`?

Reply:
good catch ... this method should be broken up into `deliverTerminalFromSource` and `safeOnComplete`